### PR TITLE
Migrate google_attachment to version 12.0

### PR DESCRIPTION
**Pull request by ddufresne**
https://isidor.numigi.net/web#id=9566&view_type=form&model=project.task&menu_id=200

### CODE REVIEW

**Review by foutoucour — APPROVED** [review]

File: google_attachment/static/src/js/AttachmentBox.js [hunk 1]
```diff
@@ -1,29 +1,51 @@
...
+ *
+ * @param {String} docName - the document name
+ * @param {String} mimeType - the document mime type
+ * @returns {String} the file name
+ */
+function buildFileNameWithExtension(docName, mimeType){
+    if(!mimeType.startsWith('application')){
+        return docName;
+    }
+    var fileTypeParts = mimeType.split('/')[1].split('.');
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 2]
```diff
@@ -32,13 +54,22 @@ function addImportFromGoogleToSideBar(clientId, apiKey){
...
             });
+
+            this._addTargetBlankToUrlLinks();
+        },
+
+        /**
+         * Open url attachments in a new browser tab.
+         */
+         _addTargetBlankToUrlLinks(){
+            this.$('.o_attachment_wrap a').attr('target', '_blank');
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 3]
```diff
@@ -68,10 +99,12 @@ function addImportFromGoogleToSideBar(clientId, apiKey){
         _createAttachmentFromGoogleDriveDoc(doc){
             var attachmentValues = {
                 name: doc.name,
+                datas_fname: buildFileNameWithExtension(doc.name, doc.mimetype),
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'datas_fname' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 4]
```diff
@@ -1,29 +1,51 @@
...
+ * @param {String} mimeType - the document mime type
+ * @returns {String} the file name
+ */
+function buildFileNameWithExtension(docName, mimeType){
+    if(!mimeType.startsWith('application')){
+        return docName;
+    }
+    var fileTypeParts = mimeType.split('/')[1].split('.');
+    var fileExtension = fileTypeParts[fileTypeParts.length - 1];
+    return docName + '.' + fileExtension;
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 5]
```diff
@@ -68,10 +99,12 @@ function addImportFromGoogleToSideBar(clientId, apiKey){
         _createAttachmentFromGoogleDriveDoc(doc){
             var attachmentValues = {
                 name: doc.name,
+                datas_fname: buildFileNameWithExtension(doc.name, doc.mimetype),
+                mimetype: doc.mimetype,
                 type: "url",
                 url: doc.url,
-                res_id: this.env.activeIds[0],
-                res_model: this.env.model,
+                res_id: this.currentResID,
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'res_id' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 6]
```diff
@@ -68,10 +99,12 @@ function addImportFromGoogleToSideBar(clientId, apiKey){
...
+                res_model: this.currentResModel,
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'res_model' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js [hunk 7]
```diff
@@ -1,29 +1,51 @@
...
+ *
+ * If the mime type is `application/json`
+ * and the file name is `My Doc`, the result will be `My Doc.json`.
+ *
+ * @param {String} docName - the document name
+ * @param {String} mimeType - the document mime type
+ * @returns {String} the file name
+ */
+function buildFileNameWithExtension(docName, mimeType){
+    if(!mimeType.startsWith('application')){
```
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Strings must use doublequote. (quotes)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js (hunk 5 above)
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'res_id' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js (hunk 3 above)
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'datas_fname' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

File: google_attachment/static/src/js/AttachmentBox.js (hunk 6 above)
Comment (ddufresne) [inline google_attachment/static/src/js/AttachmentBox.js]: Codacy Issue found: [Identifier 'res_model' is not in camel case. (camelcase)](https://app.codacy.com/app/numigi/odoo-web-addons/pullRequest?prid=3183161)

**Pull request merged**